src/octoprint/util/comm.py

```diff
@@ -2036,6 +2091,9 @@ def _readline(self):
 
 		if ret != "":
 			try:
+				if self.isSendingFileToSDWithSoftwareFlow() and self._use_xonxoff_workaround:
+					ret = self._filterXonXoffCharacters(ret)
```

Comment:
In my tests I noticed that even when NOT using the workaround but native driver mode, I got the control chars in the stream, making it necessary to always filter. Can you confirm or deny that?

Reply:
if in the firmware XON/XOFF is enabled, whenever the buffer is at 12% full it will send the chars, blindly, there is no handshake to confirm the host will interpret the chars. So if flow control is enabled in the firmware and the workaround is "not" active (the filter function) or flow control is "not" enabled from the driver, the chars will be delivered to the application level...